src/routes/sse.ts

```diff
@@ -0,0 +1,123 @@
+import { type Request, type Response, Router } from "express"
+import { instance as createLogger } from "../utils/logger"
+import { validateBroadcastBody, validateSendRequest } from "../validation/sse"
+
+const router = Router()
+const logger = createLogger("routes.sse")
+
+const connections = new Map<string, Response>()
+const HEARTBEAT_INTERVAL_MS = 30000 //30seg
+
+
+const generateConnectionId = (): string => {
+  return `conn_${Date.now()}_${Math.random().toString(36).substr(2, 9)}`
+}
+
+
+router.get("/sse/connect", (req: Request, res: Response) => {
+
+  res.writeHead(200, {
+    "Content-Type": "text/event-stream",
+    "Cache-Control": "no-cache",
+    "Connection": "keep-alive",
+    "Access-Control-Allow-Origin": "*",
+    "Access-Control-Allow-Headers": "Cache-Control"
+  })
+
+
+  const connectionId = generateConnectionId()
+
+  connections.set(connectionId, res)
+
+
+  res.write(`data: ${JSON.stringify({
+    type: "connected",
+    connectionId,
+    message: "SSE connection established",
+    timestamp: new Date().toISOString()
+  })}\n\n`)
+
+
+  req.on("close", () => {
+    connections.delete(connectionId)
+    try { clearInterval(heartbeat) } catch { }
+    logger.info("Client disconnected", { connectionId })
+  })
+
+  const heartbeat = setInterval(() => {
+    if (connections.has(connectionId)) {
+      res.write(`data: ${JSON.stringify({
+        type: "heartbeat",
+        timestamp: new Date().toISOString()
+      })}\n\n`)
+    } else {
+      clearInterval(heartbeat)
+    }
+  }, HEARTBEAT_INTERVAL_MS)
+
+  logger.info("SSE connection established", { connectionId })
+})
+
+router.post("/sse/broadcast", validateBroadcastBody, (req: Request, res: Response) => {
+  const { message, type = "notification" } = req.body as { message: string, type?: string }
```

Comment:
Avoid setting types inline.

Create a module in types/sse.ts all the types required by sse and import it and reference like `const data: MyType = req.body`

Also, your validation middleware can act not only as validator but as a sanitizer, so you can define default values on him.

Talk to Rodrigo to see how they did it in Petopia. The idea is that your validator will normalize any parameters sent in the route and put them into a `req.input` field; that way, your controller won't need to know if the data came from path parameters, body, querystring, etc.